security-admin/src/main/java/org/apache/ranger/elasticsearch/ElasticSearchAccessAuditsService.java

```diff
@@ -0,0 +1,390 @@
+		searchFields.add(new SearchField("requestData", "reqData", DATA_TYPE.STRING,
+				SEARCH_TYPE.PARTIAL));
+		searchFields.add(new SearchField("resourcePath", "resource", DATA_TYPE.STRING,
+				SEARCH_TYPE.PARTIAL));
+		searchFields.add(new SearchField("clientIP", "cliIP",
+				DATA_TYPE.STRING, SEARCH_TYPE.FULL));
+
+		searchFields.add(new SearchField("auditType", "logType",
+				DATA_TYPE.INTEGER, SEARCH_TYPE.FULL));
+		searchFields.add(new SearchField("accessResult", "result",
+				DATA_TYPE.INTEGER, SEARCH_TYPE.FULL));
+		// searchFields.add(new SearchField("assetId", "obj.assetId",
+		// SearchField.DATA_TYPE.INTEGER, SearchField.SEARCH_TYPE.FULL));
+		searchFields.add(new SearchField("policyId", "policy",
+				DATA_TYPE.INTEGER, SEARCH_TYPE.FULL));
+		searchFields.add(new SearchField("repoType", "repoType",
+				DATA_TYPE.INTEGER, SEARCH_TYPE.FULL));
+                searchFields.add(new SearchField("-repoType", "-repoType",
+        DATA_TYPE.INTEGER, SEARCH_TYPE.FULL));
+                searchFields.add(new SearchField("-requestUser", "-reqUser",
+        		DATA_TYPE.STRING, SEARCH_TYPE.FULL));
+		searchFields.add(new SearchField("resourceType", "resType",
+				DATA_TYPE.STRING, SEARCH_TYPE.FULL));
+		searchFields.add(new SearchField("reason", "reason",
+				DATA_TYPE.STRING, SEARCH_TYPE.FULL));
+		searchFields.add(new SearchField("action", "action",
+				DATA_TYPE.STRING, SEARCH_TYPE.FULL));
+
+		searchFields.add(new SearchField("startDate", "evtTime",
+				DATA_TYPE.DATE, SEARCH_TYPE.GREATER_EQUAL_THAN));
+		searchFields.add(new SearchField("endDate", "evtTime", DATA_TYPE.DATE,
+				SEARCH_TYPE.LESS_EQUAL_THAN));
+
+		searchFields.add(new SearchField("tags", "tags", DATA_TYPE.STRING, SEARCH_TYPE.PARTIAL));
+		searchFields.add(new SearchField("cluster", "cluster",
+				DATA_TYPE.STRING, SEARCH_TYPE.FULL));
+		searchFields.add(new SearchField("zoneName", "zoneName",
+				DATA_TYPE.STR_LIST, SEARCH_TYPE.FULL));
+		searchFields.add(new SearchField("agentHost", "agentHost",
+				DATA_TYPE.STRING, SEARCH_TYPE.PARTIAL));
+
+		sortFields.add(new SortField("eventTime", "evtTime", true,
+				SORT_ORDER.DESC));
+	}
+
+
+	public VXAccessAuditList searchXAccessAudits(SearchCriteria searchCriteria) {
+
+		RestHighLevelClient client = elasticSearchMgr.getClient();
+		final boolean hiveQueryVisibility = PropertiesUtil.getBooleanProperty("ranger.audit.hive.query.visibility", true);
+		if (client == null) {
+			LOGGER.warn("ElasticSearch client is null, so not running the query.");
+			throw restErrorUtil.createRESTException(
+					"Error connecting to search engine",
+					MessageEnums.ERROR_SYSTEM);
+		}
+		List<VXAccessAudit> xAccessAuditList = new ArrayList<VXAccessAudit>();
+		Map<String, Object> paramList = searchCriteria.getParamList();
+		updateUserExclusion(paramList);
+
+		SearchResponse response;
+		try {
+			response = elasticSearchUtil.searchResources(searchCriteria, searchFields, sortFields, client);
```

Comment:
Seems except  for this, everything seems to be common between Solr and Elastic. Should consider abstracting a class with common methods? E.g. updateUserExclusion(), getExcludeUsersList(), getServiceUserList(), etc. This will be easy to maintain going forward.